### PR TITLE
Feature/cds 93 segments

### PR DESCRIPTION
**Pull request by vfarah-if**
### Checklist

* [ X] If this is a releasable change, has a news fragment been added?

  <details>
  <summary>Explanation</summary>
  
  A news fragment is required for any releasable change (i.e. code that runs in or affects production) so that a corresponding changelog entry is added when releasing.
  
  Check [changelog/README.md](https://github.com/uktrade/data-hub-api/blob/master/changelog/README.md) for instructions.
  
  </details>
  
* [ X] Has this branch been rebased on top of the current `develop` branch?

  <details>
  <summary>Explanation</summary>
  
  The branch should not be stale or have conflicts at the time reviews are requested.
  
  </details>

* [ X] Is the CircleCI build passing?

### General points

<details>
<summary><strong>Other things to check</strong></summary><p></p>

* Make sure `fixtures/test_data.yaml` is maintained when updating models
* Consider the admin site when making changes to models
* Use select-/prefetch-related field lists in views and search apps, and update them when fields are added
* Make sure the README is updated e.g. when adding new environment variables

</details>

See [docs/CONTRIBUTING.md](https://github.com/uktrade/data-hub-api/blob/develop/docs/CONTRIBUTING.md) for more guidelines.

### CODE REVIEW

File: datahub/company/migrations/0109_company_segment.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+    dependencies = [
+        ('company', '0108_create_company_state_fk_base_model'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='company',
+            name='segment',
+            field=models.CharField(blank=True, choices=[(None, 'No segment or not known'), ('hep', 'HEP'), ('non-hep', 'NON-HEP')], help_text='TODO: Get from Cathy', max_length=255, null=True),
```
Comment (vfarah-if) [inline datahub/company/migrations/0109_company_segment.py:16]: Get Help Text from Cathy

File: datahub/company/migrations/0110_company_sub_segment.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        ),
+        migrations.AlterField(
+            model_name='company',
+            name='segment',
+            field=models.CharField(blank=True, choices=[(None, 'No segment or not known'), ('hep', 'HEP'), ('non-hep', 'Non-HEP')], help_text='TODO: Get from Cathy', max_length=255, null=True),
```
Comment (vfarah-if) [inline datahub/company/migrations/0110_company_sub_segment.py]: Sae with this

**Review by vfarah-if — COMMENTED** [review]
Removed the TODO's

**Review by rachaelcodes — APPROVED** [review]
Approved pending a discussion (currently in Slack) on what should be done about the Django Doctor comments.

File: datahub/company/models/company.py [hunk 3]
```diff
@@ -82,6 +82,39 @@ class GreatProfileStatus(models.TextChoices):
 
         __empty__ = 'No profile or not known'
 
+    class Segment(models.TextChoices):
+        HEP = ('hep', 'HEP')
+        NON_HEP = ('non-hep', 'Non-HEP')
```
Comment (mforner13) [inline datahub/company/models/company.py:87]: Some `help_text` here could help others understand what HEP stands for

File: datahub/company/models/company.py [hunk 4]
```diff
@@ -82,6 +82,39 @@ class GreatProfileStatus(models.TextChoices):
...
+                                    'Reassure: Change the Game')
+        PROMOTE_DEVELOP_EXPORT_CAPABILITY = ('promote_develop_export_capability',
+                                             'Promote: Develop Export Capability')
+        PROMOTE_COMMUNICATE_BENEFITS = ('promote_communicate_benefits',
+                                        'Promote: Communicate Benefits')
+        PROMOTE_CHANGE_THE_GAME = ('promote_change_the_game',
+                                   'Promote: Change the Game')
+        CHALLENGE = ('challenge', 'Challenge')
+
+        __empty__ = 'No Sub-Segment or not known'
```
Comment (mforner13) [inline datahub/company/models/company.py:116]: Will this option ever be used if the field is allowed to be blank?

**Review by mforner13 — COMMENTED** [review]
Looks good! Just two small comments 😃

File: datahub/company/models/company.py (hunk 3 above)
Comment (vfarah-if) [reply datahub/company/models/company.py:87]: Yes we are waiting on Cathy for that - which is why we removed the TODO :) Will update that in the next feature when she gets back to us

File: datahub/company/models/company.py [hunk 5]
```diff
@@ -82,6 +82,39 @@ class GreatProfileStatus(models.TextChoices):
 
         __empty__ = 'No profile or not known'
 
+    class Segment(models.TextChoices):
```
Comment (elcct) [inline datahub/company/models/company.py:85]: The name seems quite generic. What does it refer to?

File: datahub/company/models/company.py [hunk 6]
```diff
@@ -82,6 +82,39 @@ class GreatProfileStatus(models.TextChoices):
...
+        HEP = ('hep', 'HEP')
+        NON_HEP = ('non-hep', 'Non-HEP')
+
+        __empty__ = 'No segment or not known'
+
+    class SubSegment(models.TextChoices):
```
Comment (elcct) [inline datahub/company/models/company.py:91]: The `SubSegment` suggests that this is somewhat related to `Segment`, but how?

Comment (vfarah-if) [reply datahub/company/models/company.py:91]: I think it is just a domain term for different types of lookup data, so you may be reading into it deeper than it needs. S

File: datahub/company/models/company.py (hunk 4 above)
Comment (vfarah-if) [reply datahub/company/models/company.py:116]: Yes it is optional data

File: datahub/company/models/company.py [hunk 7]
```diff
@@ -296,6 +329,18 @@ class GreatProfileStatus(models.TextChoices):
         help_text='Last updated from D&B',
         db_index=True,
     )
+    segment = models.CharField(
```
[the remark this answers is not in the document]
Comment (vfarah-if) [reply datahub/company/models/company.py:332]: Leaving this as is

File: datahub/company/models/company.py [hunk 8]
```diff
@@ -296,6 +329,18 @@ class GreatProfileStatus(models.TextChoices):
...
+        max_length=MAX_LENGTH,
+        null=True,
+        blank=True,
+        choices=Segment.choices,
+    )
+    sub_segment = models.CharField(
```
[the remark this answers is not in the document]
Comment (vfarah-if) [reply datahub/company/models/company.py:338]: Leaving this as is

File: datahub/company/models/company.py (hunk 5 above)
Comment (vfarah-if) [reply datahub/company/models/company.py:85]: Segment refers to Segment data. More information can be seen here - https://uktrade.atlassian.net/jira/software/projects/CDS/boards/223?selectedIssue=CDS-93

Comment (vfarah-if) [reply datahub/company/models/company.py:85]: Let me know what name you would suggest for Segment and Subsegment references?

File: datahub/company/models/company.py (hunk 6 above)
Comment (vfarah-if) [reply datahub/company/models/company.py]: Changed to ExportSubSegment

File: datahub/company/models/company.py (hunk 5 above)
Comment (vfarah-if) [reply datahub/company/models/company.py]: Changed to ExportSubSegment and ExportSegment

File: datahub/company/models/company.py (hunk 3 above)
Comment (vfarah-if) [reply datahub/company/models/company.py]: The value stayed the same but the name is **High Priority Export**

**Review by mforner13 — APPROVED** [review]

**Pull request merged**